### ISSUE TITLE
Machine Learning with Scikit-Learn

### ISSUE THREAD

**Issue by WayneLambert** [WayneLambert/notebook#21]
<img width="1244" alt="2020-12-06_13-35-21" src="https://user-images.githubusercontent.com/44729076/101281629-ff872a80-37c7-11eb-8705-c2c0f0465003.png">

## Scikit Learn

Scikit-learn is the most populalar open-source machine learning library for Python. It strives to have a uniform interface across all methods and algorithms.

Given a scikit-learn estimator object named model, the following methods are available:

`model.fit()` which is used to fit training data

#### Supervised Learning Applications

`model.fit()` accepts two arguments: the data **X** and the labels **y**. The method therefore takes the form of:

`model.fit(X, y)`

#### Unsupervised Learning Applications

`model.fit()` accepts a single argument, the data **X**. The method in this case therefore takes the form of:

``model.fit(X)`

#### Supervised Estimators

The following methods are available for supervised estimator objects:

##### model.predict()

`model.predict()`: given a trained model, predicts the label of a new set of data. This method accepts just hte one argument, the new data **X_new** which is an array object. The method therefore takes the form:

`model.predict(X_new)` and returns the learned label for each object in the array.

##### model.proba()

`model.proba()` is used for classification problems. This method returns the probability that a new observation has each categorical label. In this scenario, the label with the highest probability is return by `model.predict()`

##### model.score()

`model.score()` is used for classification problems. This method returns the probability that a new observation has each categorical label. In this scenario, the label with the highest probability is return by `model.predict()`
